### PR TITLE
Clarify which dependencies are updated by default

### LINKED ISSUE #20454: By default, dependabot only updates direct dependencies
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.

### What part(s) of the article would you like to see updated?

This sentence should be clarified

>  By default all dependencies that are explicitly defined in a manifest or lock file are kept up to date.

Lock files usually keep both direct and indirect dependencies, but dependabot only updates the former by default.

### Additional information

I will create a PR :)

### PR DESCRIPTION
**Pull request by deivid-rodriguez**
By default, dependabot only keeps direct dependencies (those explicitly defined in a manifest file) up to date, except for security PRs.

### Why:

Closes #20454.

### What's being changed (if available, include any code snippets, screenshots, or gifs):

I'm clarifying a sentence about which dependencies dependabot updates by default.

### Check off the following:

- [ ] I have reviewed my changes in staging (look for the "Automatically generated comment" and click the links in the "Preview" column to view your latest changes).
- [ ] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

**Comment by cmwilson21** [comment]
@deivid-rodriguez Thanks for opening a PR and linking it to your issue! ✨ 

I'll get this triaged for review ⚡

**Review by felicitymay — DISMISSED** [review]
Thanks for this change 💖

**Comment by felicitymay** [comment]
@deivid-rodriguez - I've just remembered that `allow` and `ignore` configuration options apply to both Dependabot security updates and version updates. I'm not confident that this change is correct for security updates as well as version updates. I'll check with the Dependabot updates team and determine whether we need to make any changes before we merge this pull request.

File: data/reusables/dependabot/default-dependencies-allow-ignore.md
Comment (deivid-rodriguez) [inline data/reusables/dependabot/default-dependencies-allow-ignore.md:1]:
```suggestion
By default all dependencies that are explicitly defined in a manifest are kept up to date, but note that if you have security updates enabled, it will also create updates for vulnerable dependencies only included in lock files. You can use `allow` and `ignore` to customize which dependencies to maintain with version updates. {% data variables.product.prodname_dependabot %} checks for all allowed dependencies and then filters out any ignored dependencies or versions. So a dependency that is matched by both an `allow` and an `ignore` will be ignored.
```

Comment (deivid-rodriguez) [reply data/reusables/dependabot/default-dependencies-allow-ignore.md:1]: I'm not sure how to best clarify this, but I made some initial suggestion!

Comment (felicitymay) [reply data/reusables/dependabot/default-dependencies-allow-ignore.md:1]: Thanks for clarifying the behavior here. I think this might be better as two sentences, how about:

```suggestion
By default all dependencies that are explicitly defined in a manifest are kept up to date by {% data variables.product.prodname_dependabot %} version updates. In addition, {% data variables.product.prodname_dependabot %} security updates also update vulnerable dependencies that are defined in lock files. You can use `allow` and `ignore` to customize which dependencies to maintain. {% data variables.product.prodname_dependabot %} checks for all allowed dependencies and then filters out any ignored dependencies or versions. So a dependency that is matched by both an `allow` and an `ignore` will be ignored.
```

Comment (deivid-rodriguez) [reply data/reusables/dependabot/default-dependencies-allow-ignore.md:1]: I think that's better, yeah!

**Review by felicitymay — APPROVED** [review]
If you're happy with the revised wording @deivid-rodriguez, I'll merge the update once I've checked the preview to make sure that I haven't added any errors to your change 🙂

**Comment by deivid-rodriguez** [comment]
Thank you!

**Pull request merged**
